mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -163,25 +190,28 @@ class StringIndexerModel (
     }
     transformSchema(dataset.schema, logging = true)
 
+    val metadata = NominalAttribute.defaultAttr
+      .withName($(outputCol)).withValues(labels).toMetadata()
```

Comment:
Yep, that's what I meant:  In ```withValues(labels)```, labels can be set as:
```
val labels = getHandleInvalid match {
  case StringIndexer.KEEP_UNSEEN_LABEL => labels :+ "__unknown"
  case _ => labels
}
```

I'm adding underscores to the attribute name to make it a little less likely to hit conflicts.